channeldb/payment_control_test.go

```diff
@@ -878,19 +880,41 @@ func TestPaymentControlMultiShard(t *testing.T) {
 			}
 		}
 
-		// If any of the two attempts settled, the payment should end
-		// up in the Succeeded state. If both failed the payment should
-		// also be Failed at this poinnt.
-		finalStatus := StatusFailed
-		if test.settleFirst || test.settleLast {
+		var (
+			finalStatus PaymentStatus
+			registerErr error
+		)
+
+		switch {
+		// If one of the attempts settled but the other failed with
+		// terminal error, this will result in an error state as
+		// clearly something went wrong.
+		case test.settleFirst && !test.settleLast:
 			finalStatus = StatusSucceeded
+			registerErr = ErrPaymentAlreadySucceeded
+
+		case !test.settleFirst && test.settleLast:
+			finalStatus = StatusInFlight
```

Comment:
so this test creates three HTLCs, where the second HTLC is always failed. For the first HTLC, we either settle it or fail the **payment**, this means after this step, the payment has reached a terminal state. And based on whether `settleFirst` is true or not, `settleLast` may give us an error(see L848).

In this specific case, because `settleFirst` is false, it means the payment has failed. But the status is inflight because the third HTLC is still inflight. And the third is inflight because, when `settledLast` is true, we will settle the HTLC and receive an error because the payment is already failed.

That being said, I will remove this test because it's too complicated.